### PR TITLE
telemetry: add pk type of clusterindex feature usage information

### PR DESCRIPTION
**Pull request by kaixu120811**
### What problem does this PR solve?

Issue Number: #24695 

Problem Summary:

### What is changed and how it works?
add pk type into cluster index feature usage information.

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
1. Setup TiDB
2. Create a CLUSTER INDEXED table with INT/NON-INT PK
3. Execute: admin show telemetry.

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Add PK type information of Cluster Indexed table: INT or NON-INT.

### CODE REVIEW

File: telemetry/data_feature_usage.go [hunk 1]
```diff
@@ -15,31 +15,60 @@ package telemetry
...
 }
 
 func getFeatureUsage(ctx sessionctx.Context) (*featureUsage, error) {
-	// init
-	usageInfo := featureUsage{
-		ClusterIndex: make(map[string]bool),
+
+	clusterIdxUsage, err := GetClusterIndexUsageInfo(ctx)
+	if err != nil {
+		logutil.BgLogger().Info(err.Error())
```
Comment (breeswish) [inline telemetry/data_feature_usage.go:41]: How about continue reporting other features when the retrieval of some of the feature usage is failed?

File: telemetry/data_feature_usage.go [hunk 2]
```diff
@@ -48,21 +77,49 @@ func getFeatureUsage(ctx sessionctx.Context) (*featureUsage, error) {
...
+			tblClusteredInfo.IsClustered = true
+			table, err := infoSchema.TableByName(model.NewCIStr(row.GetString(2)), model.NewCIStr(row.GetString(3)))
+			if err != nil {
+				continue
+			}
+			tableInfo := table.Meta()
+			if tableInfo.PKIsHandle {
+				tblClusteredInfo.ClusterPKType = "INT"
+			} else if tableInfo.IsCommonHandle {
+				tblClusteredInfo.ClusterPKType = "NON_INT"
```
Comment (breeswish) [inline telemetry/data_feature_usage.go:113]: Is it possible to query this information from the information_schema?

/cc @lzmhhh123  @lysu PTAL

Comment (lzmhhh123) [reply telemetry/data_feature_usage.go:113]: It doesn't matter. The SQL accessing `information_schema` reads from information schema too.

**Review by lzmhhh123 — APPROVED** [review]

File: telemetry/data_feature_usage.go (hunk 2 above)
Comment (lysu) [reply telemetry/data_feature_usage.go:113]: +1 information_schema has hidden int detail for the user as PM's design, so we only can read it from tableInfo

**Review by kaixu120811 — COMMENTED** [review]
Answer review questions.

**Pull request merged**